src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEModuleSymbol.cs

```diff
@@ -641,47 +641,55 @@ internal NamedTypeSymbol LookupTopLevelMetadataType(ref MetadataTypeName emitted
         /// <param name="fullName">Type to look up.</param>
-        /// <returns>Assembly symbol or null.</returns>
+        /// <returns>A tuple of the forwarded to assemblies.</returns>
         /// <remarks>
-        /// The returned assembly may also forward the type.
+        /// The returned assemblies may also forward the type.
         /// </remarks>
-        internal AssemblySymbol GetAssemblyForForwardedType(ref MetadataTypeName fullName)
+        internal (AssemblySymbol FirstSymbol, AssemblySymbol SecondSymbol) GetAssembliesForForwardedType(ref MetadataTypeName fullName)
         {
             string matchedName;
-            AssemblyReferenceHandle assemblyRef = Module.GetAssemblyForForwardedType(fullName.FullName, ignoreCase: false, matchedName: out matchedName);
-            return assemblyRef.IsNil ? null : this.GetReferencedAssemblySymbol(assemblyRef);
+            (int FirstIndex, int SecondIndex) indices = this.Module.GetAssemblyRefsForForwardedType(fullName.FullName, ignoreCase: false, matchedName: out matchedName);
+
+            var firstSymbol = indices.FirstIndex < 0 ? null : GetReferencedAssemblySymbol(indices.FirstIndex);
+            var secondSymbol = indices.SecondIndex < 0 ? null : GetReferencedAssemblySymbol(indices.SecondIndex);
+
+            return (firstSymbol, secondSymbol);
         }
 
         internal IEnumerable<NamedTypeSymbol> GetForwardedTypes()
         {
-            foreach (KeyValuePair<string, AssemblyReferenceHandle> forwarder in Module.GetForwardedTypes())
+            foreach (KeyValuePair<string, (int FirstIndex, int SecondIndex)> forwarder in Module.GetForwardedTypes())
             {
-                var assemblySymbol = this.GetReferencedAssemblySymbol(forwarder.Value);
-                if ((object)assemblySymbol == null)
+                if (forwarder.Value.FirstIndex < 0)
                 {
                     continue;
                 }
+
                 var name = MetadataTypeName.FromFullName(forwarder.Key);
-                yield return assemblySymbol.LookupTopLevelMetadataType(ref name, digThroughForwardedTypes: true);
-            }
-        }
+                AssemblySymbol firstSymbol = this.GetReferencedAssemblySymbol(forwarder.Value.FirstIndex);
 
-        private AssemblySymbol GetReferencedAssemblySymbol(AssemblyReferenceHandle assemblyRef)
-        {
-            int referencedAssemblyIndex;
-            try
-            {
-                referencedAssemblyIndex = Module.GetAssemblyReferenceIndexOrThrow(assemblyRef);
-            }
-            catch (BadImageFormatException)
-            {
-                return null;
+                if ((object)firstSymbol == null)
+                {
+                    continue;
+                }
+
+                if (forwarder.Value.SecondIndex >= 0)
+                {
+                    var secondSymbol = this.GetReferencedAssemblySymbol(forwarder.Value.SecondIndex);
+
+                    if ((object)secondSymbol != null)
```

Comment:
Are we testing the case where there are multiple forwards but the second referenced assembly is missing? Shouldn't we `continue` in that case rather than `yield return firstSymbol;`, to match the case when the first referenced assembly is missing?